zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/FormControllerTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import io.camunda.service.FormServices;
+import io.camunda.service.entities.FormEntity;
+import io.camunda.service.exception.NotFoundException;
+import io.camunda.service.security.auth.Authentication;
+import io.camunda.zeebe.gateway.rest.RestControllerTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest(value = FormController.class, properties = "camunda.rest.query.enabled=true")
+public class FormControllerTest extends RestControllerTest {
+
+  private static final Long VALID_FORM_KEY = 1L;
+  private static final Long INVALID_FORM_KEY = 999L;
+  private static final String FORM_ITEM_JSON =
+      """
+      {
+        "key": 1,
+        "tenantId": "tenant-1",
+        "bpmnId": "bpmn-1",
+        "schema": "schema",
+        "version": 1
+      }
+      """;
+
+  @Autowired private MockMvc mockMvc;
```

Comment:
❌ The `RestControllerTest` already provides a `WebClient` you should be able to use in the tests.